### PR TITLE
Clip soundfield into float64 range for plotting

### LINKED ISSUE #50: Monofrequent dipole source example plot
The plot of the mono-frequent dipole source has a (normalization?) issue in the current version.

frequency-domain-9_new

http://python.sfstoolbox.org/en/latest/frequency-domain.html#sfs.mono.source.point_dipole

Before it was

frequency-domain-9_old

http://python.sfstoolbox.org/en/0.3.1/#sfs.mono.source.point_dipole

### PR DESCRIPTION
**Pull request by trettberg**
closes #50 

(This may become obsolete in matplotlib 2.2 or not.
Since it's easy enough, I propose to fix it now anyway.)

### CODE REVIEW

**Comment by hagenw** [comment]
I'm also in favor of including a fix for this.
An alternative solution would be to restrict matplotlib to <=2.0.2 in the dependencies of sfs and change this to >=2.2 when they have fixed matplotlib. I'm not sure what would be the better option.

**Comment by mgeier** [comment]
I wouldn't restrict the `matplotlib` version, as this could be an unnecessary inconvenience for users.
BTW, currently we don't even officially depend on `matplotlib` (according to `setup.py`).

The fix suggested by @trettberg seems less intrusive, but what about limiting the fix to the affected versions (since it looks like it'll be fixed in 2.2.0)?
Something like:

```python
import matplotlib
if matplotlib.__version__.startswith('2.1'):
    p = np.clip(p, -1e15, 1e15)
```

File: sfs/plot.py [hunk 1]
```diff
@@ -284,6 +285,9 @@ def soundfield(p, grid, xnorm=None, cmap='coolwarm_clip', vmin=-2.0, vmax=2.0,
     if ax is None:
         ax = plt.gca()
 
+    if matplotlib_version.startswith('2.1.'):
+        p = np.clip(p, -1e15, 1e15)  # clip to float64 range
```
Comment (hagenw) [inline sfs/plot.py]: Maybe we add a link to the matplotlib bug (https://github.com/matplotlib/matplotlib/issues/10567) on the line before to highlight why we are doing this?

Comment (hagenw) [reply sfs/plot.py]: Otherwise this implementation is fine with me.

**Pull request merged**